### PR TITLE
Feat/dodont fixes

### PR DESCRIPTION
**Pull request by fulopdaniel**
Fix various issues:

- Icon was clipped in the settings sidebar
- Focus was not working correctly
- On link chooser cancel click the next position of the modal was at the top left
- On textarea input the dragging was not working correctly

### CODE REVIEW

File: packages/dos-donts-block/src/settings.ts [hunk 1]
```diff
@@ -269,6 +269,7 @@ export const settings = defineSettings({
             id: 'imagesSection',
```
Comment (ragi96) [inline packages/dos-donts-block/src/settings.ts:269]: I think this empty section shouldn't be displayed on the initial release

Screenshot 2023-07-25 at 13 05 23

File: packages/dos-donts-block/src/DosDontsBlock.tsx [hunk 2]
```diff
@@ -200,23 +200,43 @@ export const DosDontsBlock: FC<BlockProps> = ({ appBridge }) => {
         setAndSaveItems(newItems);
     };
 
+    const onChangeLocalItem = (
+        itemId: string,
+        value: string | number | undefined,
```
Comment (ragi96) [inline packages/dos-donts-block/src/DosDontsBlock.tsx:205]:
```suggestion
        value: values,
```

add a new type for this union so we can get rid of the repetition

File: packages/dos-donts-block/src/DosDontsBlock.tsx [hunk 3]
```diff
@@ -200,23 +200,43 @@ export const DosDontsBlock: FC<BlockProps> = ({ appBridge }) => {
...
+        type: 'title' | 'body' | 'type' | 'imageId'
```
Comment (ragi96) [inline packages/dos-donts-block/src/DosDontsBlock.tsx:206]:
```suggestion
        type: types
```
add a new type for this union so we can get rid of the repetition

File: packages/dos-donts-block/src/settings.ts (hunk 1 above)
Comment (fulopdaniel) [reply packages/dos-donts-block/src/settings.ts:269]: Good catch, fixed

File: packages/dos-donts-block/src/DosDontsBlock.tsx (hunk 3 above)
Comment (fulopdaniel) [reply packages/dos-donts-block/src/DosDontsBlock.tsx]: Thanks, fixed and the other one as well.

File: packages/dos-donts-block/src/types.ts [hunk 4]
```diff
@@ -77,6 +77,11 @@ export type SortableDoDontItemProps = {
     doColor: Color;
     dontColor: Color;
     onChangeItem: (id: string, value: string | number | undefined, type: 'title' | 'body' | 'type' | 'imageId') => void;
+    onChangeLocalItem: (
+        id: string,
+        value: string | number | undefined,
+        type: 'title' | 'body' | 'type' | 'imageId'
+    ) => void;
```
Comment (ragi96) [inline packages/dos-donts-block/src/types.ts:84]:
```suggestion
    onChangeLocalItem: (id: string, value: ValueType, type: ChangeType) => void;
```

Also on the other types

Comment (fulopdaniel) [reply packages/dos-donts-block/src/types.ts]: Thanks, fixed!

File: packages/dos-donts-block/src/types.ts [hunk 5]
```diff
@@ -76,7 +76,8 @@ export type SortableDoDontItemProps = {
     style: DoDontStyle;
     doColor: Color;
     dontColor: Color;
-    onChangeItem: (id: string, value: string | number | undefined, type: 'title' | 'body' | 'type' | 'imageId') => void;
+    onChangeItem: (id: string, value: string | number | undefined, type: ChangeType) => void;
```
Comment (ragi96) [inline packages/dos-donts-block/src/types.ts:79]:
```suggestion
    onChangeItem: (id: string, value: ValueType, type: ChangeType) => void;
```

File: packages/dos-donts-block/src/types.ts [hunk 6]
```diff
@@ -76,7 +76,8 @@ export type SortableDoDontItemProps = {
...
+    onChangeLocalItem: (id: string, value: string | number | undefined, type: ChangeType) => void;
```
Comment (ragi96) [inline packages/dos-donts-block/src/types.ts:80]:
```suggestion
    onChangeLocalItem: (id: string, value: ValueType, type: ChangeType) => void;
```

**Review by ragi96 — APPROVED** [review]

**Pull request merged**